### PR TITLE
feat: [IOPAE-543] be auth using next-auth

### PR DESCRIPTION
**Pull request by rodoviti**
Use next-auth auth functionality to create/verify JWT session cookie and to provide an authorization layer on top of the route hadlers. Moreover, "inject" logged user details to every protected route handlers by wrapping them.

#### List of Changes

- create a next-auth (JWT) session
- add route handler wrapper `withJWTAuthHandler`, this will protect the routes which require authentication and inject the logged user information to them
- wrap all `io-services-cms` related route handlers
- Add custom claims `parameters` in `BackOfficeUser`

#### Motivation and Context

These changes are required in order to secure backend API

#### Screenshots (if appropriate):

https://github.com/pagopa/io-services-cms/assets/7738875/0f165fc3-adf3-47fa-a077-bafaebc60b66

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/backoffice/src/app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -31,37 +32,55 @@ const authOptions: NextAuthOptions = {
...
-          session_token: string;
-        };
+        const {
+          identity_token: identity_token
+        } = credentials as SelfCareIdentity;
 
         // first quick token check
-        if (isNullUndefinedOrEmpty(session_token)) {
-          throw new Error("null undefined or empty session token");
+        if (isNullUndefinedOrEmpty(identity_token)) {
```
Comment (gunzip) [inline apps/backoffice/src/app/api/auth/[...nextauth]/route.ts:40]: why not `NonEmptyString.is` ?

File: apps/backoffice/src/app/api/utils/io-services-cms-proxy.ts [hunk 2]
```diff
@@ -37,31 +38,26 @@ export const forwardIoServicesCmsRequest = (
...
     body: jsonBody,
-    "x-user-email": "SET_RETRIEVED_USER_EMAIL_HERE", // TODO: replace with real value
-    // "x-user-groups": "SET_RETRIEVED_USER_GROUPS_HERE", // TODO: replace with real value
-    "x-user-id": "SET_RETRIEVED_USER_ID_HERE", // TODO: replace with real value
-    "x-subscription-id": "SET_RETRIEVED_SUBSCRIPTION_ID_HERE", // TODO: replace with real value
-    "x-user-groups": "SET_RETRIEVED_USER_GROUPS_HERE" // TODO: replace with real value
+    "x-user-email": headerParameters.userEmail,
+    "x-user-id": headerParameters.userId,
+    "x-subscription-id": headerParameters.subscriptionId,
+    "x-user-groups": headerParameters.userGroups.toString()
```
Comment (rodoviti) [inline apps/backoffice/src/app/api/utils/io-services-cms-proxy.ts:56]: **chore:** use join to "comma-separate" values
```suggestion
    "x-user-groups": headerParameters.userGroups.join(",")
```

**Review by rodoviti — COMMENTED** [review]
**question (non-blocking):** what do you think to move `handler-wrappers.ts` from `apps/backoffice/src/lib/` to `apps/backoffice/src/app/api/lib/`?

File: apps/backoffice/src/app/api/auth/[...nextauth]/route.ts (hunk 1 above)
Comment (giamma1295) [reply apps/backoffice/src/app/api/auth/[...nextauth]/route.ts]: 08b247d16c55a3f2434bb556d5c8508f8fbddbf1

File: apps/backoffice/src/app/api/utils/io-services-cms-proxy.ts (hunk 2 above)
Comment (giamma1295) [reply apps/backoffice/src/app/api/utils/io-services-cms-proxy.ts]: cffb580ea368f32b910eb6695aff4dad46da7846

**Review by giuseppedipinto — APPROVED** [review]
LGTM!

**Pull request merged**